src/com/ms/silverking/cloud/dht/daemon/storage/convergence/management/SyncController.java

```diff
@@ -321,189 +325,191 @@ boolean serviceCompletionQueue(long timeout, TimeUnit unit) {
+          alertSW.reset();
+        } else {
+          if (alertSW.getSplitSeconds() > nonCompletionAlertThresholdSeconds && !incompleteDisplayed) {
+            incompleteDisplayed = true;
+            displayIncomplete();
+          }
+        }
+        if (resendSW.getSplitSeconds() > resendCheckIntervalSeconds) {
+          checkForResends();
+          resendSW.reset();
+        }
+      }
+      if (hasErrors) {
+        throw new ConvergenceException("SyncController unable to complete due to errors");
+      } else if (abandoned) {
+        throw new ConvergenceException("Abandoned");
+      } else if (!complete()) {
+        throw new ConvergenceException("Sync timed out");
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
 
-    private void computeStatus() {
-        ineligibleActionsSize = ineligibleActions.size();
-        eligibleSyncsSize = eligibleSyncs.size();
-        activeSyncsSize = activeSyncs.size();
-        totalOutstandingSyncRetrievalRequests = computeTotalOutstandingSyncRetrievalRequests();
+  private void displayIncomplete() {
+    Log.warningAsync("Incomplete requests");
+    for (Map.Entry<UUIDBase, ReplicaSyncRequest> e : activeSyncs.entrySet()) {
+      Log.warningAsyncf("%s\t%s", e.getKey(), e.getValue());
     }
+    Log.warningAsync("End incomplete requests");
+  }
 
-    // lock must be held
-    public void abandon() {
-        lock.lock();
-        try {
-               eligibleSyncs.clear();
-               abandoned = true;
-        } finally {
-            lock.unlock();
-        }
-    }
-    
-    public double elapsedSeconds() {
-        return elapsedSW.getSplitSeconds();
-    }
-    
-    public String getStatus() {
-        return String.format("%s:%s:%s:%s:%s", ineligibleActionsSize, eligibleSyncsSize, activeSyncsSize, totalOutstandingSyncRetrievalRequests, ineligibleActionsSize + eligibleSyncsSize);
-    }
+  private long computeTotalOutstandingSyncRetrievalRequests() {
+    long total;
 
-    public void updateProgress(UUIDBase uuid, Pair<Long, Long> progress) {
-        outstandingSyncRetrievalRequests.put(uuid, progress.getV1().intValue());
+    total = 0;
+    for (Integer x : outstandingSyncRetrievalRequests.values()) {
+      total += x;
     }
- }
+    return total;
+  }
+
+  private void computeStatus() {
+    ineligibleActionsSize = ineligibleActions.size();
+    eligibleSyncsSize = eligibleSyncs.size();
+    activeSyncsSize = activeSyncs.size();
+    totalOutstandingSyncRetrievalRequests = computeTotalOutstandingSyncRetrievalRequests();
+  }
+
+  // lock must be held
+  public void abandon() {
+    lock.lock();
+    try {
+      eligibleSyncs.clear();
+      abandoned = true;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public double elapsedSeconds() {
+    return elapsedSW.getSplitSeconds();
+  }
+
+  public String getStatus() {
+    return String.format("%s:%s:%s:%s:%s", ineligibleActionsSize, eligibleSyncsSize, activeSyncsSize,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: ineligibleActionsSize, eligibleSyncsSize, &#39;+&#39; expression, activeSyncsSize. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 508. Consider formatting this data with &#39;%d&#39; instead.